tempodb/encoding/vparquet/block_search.go

```diff
@@ -0,0 +1,404 @@
+package vparquet
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"math"
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+	tempo_io "github.com/grafana/tempo/pkg/io"
+	pq "github.com/grafana/tempo/pkg/parquetquery"
+	"github.com/grafana/tempo/pkg/tempopb"
+	v1 "github.com/grafana/tempo/pkg/tempopb/trace/v1"
+	"github.com/grafana/tempo/pkg/util"
+	"github.com/grafana/tempo/tempodb/backend"
+	"github.com/grafana/tempo/tempodb/encoding/common"
+	"github.com/opentracing/opentracing-go"
+	"github.com/segmentio/parquet-go"
+)
+
+// These are reserved search parameters
+const (
+	LabelName   = "name"
+	LabelStatus = "status"
+
+	StatusCodeTag   = "status.code"
+	StatusCodeUnset = "unset"
+	StatusCodeOK    = "ok"
+	StatusCodeError = "error"
+)
+
+var StatusCodeMapping = map[string]int{
+	StatusCodeUnset: int(v1.Status_STATUS_CODE_UNSET),
+	StatusCodeOK:    int(v1.Status_STATUS_CODE_OK),
+	StatusCodeError: int(v1.Status_STATUS_CODE_ERROR),
+}
+
+type BackendReaderAt struct {
```

Comment:
We have a similar idea here:

https://github.com/grafana/tempo/blob/91673c8ccc21aa6724853da161ba030573586d5a/tempodb/backend/readerat.go#L29-L36

Unsure if that can be reused or not.